src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -853,18 +855,28 @@ public synchronized InputStream getInputStream(ZipEntry ze)
 
         // wrap a verifier stream around the real stream
         return new JarVerifier.VerifierStream(
-            getManifestFromReference(),
-            verifiableEntry(ze),
-            super.getInputStream(ze),
-            jv);
+                getManifestFromReference(),
+                verifiableEntry(ze),
+                super.getInputStream(ze),
+                jv);
+
     }
 
-    private JarEntry verifiableEntry(ZipEntry ze) {
+    private JarEntry verifiableEntry(ZipEntry ze) throws ZipException {
         if (ze instanceof JarFileEntry) {
             // assure the name and entry match for verification
             return ((JarFileEntry)ze).realEntry();
         }
-        ze = getJarEntry(ze.getName());
+        // ZipEntry::getName should not return null
+        if(ze.getName() != null) {
+            ze = getJarEntry(ze.getName());
+        } else {
+            throw new ZipException("Error: ZipEntry::getName returned null!");
```

Comment:
I'd probably leave out the "Error:" and the "!".

Reply:
OK, I can do that if you prefer.